### ISSUE TITLE
TimeoutException should handled during `send()` not in `_parse_reponse`

### ISSUE THREAD

**Issue by waynerv** [ramnes/notion-sdk-py#61]
```
def _parse_response(self, response: Response) -> Any:
        try:
            response.raise_for_status()
        except httpx.TimeoutException as error:
            if is_timeout_error(error):
                raise RequestTimeoutError()
            raise
        except httpx.HTTPStatusError as error:
            body = error.response.json()
            code = body.get("code")
            if is_api_error_code(code):
                body = APIErrorResponseBody(code=code, message=body["message"])
                raise APIResponseError(error.response, body)
            raise HTTPResponseError(error.response)

        return response.json()
```
Timeout exception handling logic above in `BaseClient._parse_response()` would never run, becuase `raise_for_status()` won't raise `TimeoutException`.

**Comment by mutt0-ds** [comment]
@ramnes I must warn you that I'm just a beginner, but I love the work you are doing and I'm trying to help whenever is possible. Please take my code with a grain of salt ahah

**Comment by waynerv** [comment]
@mutt0-ds You should make a pull request, this commit is still in your forked repo. 

Just remove this codes is not enough, Timeout Error should be properly handled.

**Comment by mutt0-ds** [comment]
Thanks for the feedback, I'll try to work on it in my testing environment and then make a pull request

**Comment by ramnes** [comment]
Hey @mutt0-ds, are you still willing to work on this one? :)

**Comment by mutt0-ds** [comment]
Hey @ramnes sorry, I'll try to work on it next week since I've been using the library for months and I know it better now!

**Comment by ramnes** [comment]
Fixed in #61
